packages/teams-js/src/internal/communication.ts

```diff
@@ -194,6 +311,58 @@ function waitForResponse<T>(requestId: number): Promise<T> {
   });
 }
 
+/**
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export function sendMessageToParentWithVersion(
+  apiVersion: string,
+  actionName: string,
+  args: any[],
+  callback?: Function,
+): void;
+
+/**
+ * @internal
+ * Limited to Microsoft-internal use
+ */
+export function sendMessageToParentWithVersion(apiVersion: string, actionName: string, callback?: Function): void;
+
+/**
+ * @hidden
+ * Send a message to parent. Uses nativeInterface on mobile to communicate with parent context
+ * Additional apiVersion parameter is added, which provides the ability to send api version number to parent
+ * for telemetry work. The code inside of this function will be used to replace sendMessageToParent function
+ * and this function will be removed when the project is completed.
+ */
+export function sendMessageToParentWithVersion(
+  apiVersion: string,
+  actionName: string,
+  argsOrCallback?: any[] | Function,
+  callback?: Function,
+): void {
+  let args: any[] | undefined;
+  if (argsOrCallback instanceof Function) {
+    callback = argsOrCallback;
+  } else if (argsOrCallback instanceof Array) {
+    args = argsOrCallback;
+  }
+
+  if (!isFollowApiVersionLabelFormat(apiVersion)) {
+    throw Error(
+      `apiVersion: ${apiVersion} passed in doesn't follow the pattern starting with 'v' followed by digits, please check.`,
+    );
+  }
+
+  // APIs with v0 represents beta changes haven't been implemented on them
```

Comment:
This comment is really helpful.. and it also makes me to think:
how do you think of the idea that having an enum for versioning? (in there for v0 we can also clarify it as beta like what you are doing here)
That may reduce of chancing that people accidentally passing an bad format of apiversion.

Reply:
Yeah that is something we can do. I think I had an idea before but then probably forget about it. Thanks for reminding and will add it.

Also we have another layer of protection that if you pass in a value doesn't following the pattern 'v(digits)', you will fail on build because there is a check when sdk request is to be sent out.

Reply:
Updated with an ApiVersion enum to set version number.